search/search_index_values.hpp

```diff
@@ -25,43 +25,69 @@
 // A wrapper around feature index.
 struct FeatureIndexValue
 {
-  FeatureIndexValue() : m_featureId(0) {}
+  FeatureIndexValue() : m_id(0) {}
 
-  FeatureIndexValue(uint64_t featureId) : m_featureId(featureId) {}
+  FeatureIndexValue(uint64_t id) : m_id(id) {}
 
-  bool operator<(FeatureIndexValue const & o) const { return m_featureId < o.m_featureId; }
+  bool operator<(FeatureIndexValue const & o) const { return m_id < o.m_id; }
 
-  bool operator==(FeatureIndexValue const & o) const { return m_featureId == o.m_featureId; }
+  bool operator==(FeatureIndexValue const & o) const { return m_id == o.m_id; }
 
-  void Swap(FeatureIndexValue & o) { swap(m_featureId, o.m_featureId); }
+  void Swap(FeatureIndexValue & o) { swap(m_id, o.m_id); }
 
-  uint64_t m_featureId;
+  uint64_t m_id;
 };
 
+namespace std
+{
+template <>
+class hash<FeatureIndexValue>
+{
+public:
+  size_t operator()(FeatureIndexValue const & value) const
+  {
+    return std::hash<uint64_t>{}(value.m_id);
+  }
+};
+}  // namespace std
+
 struct FeatureWithRankAndCenter
 {
-  FeatureWithRankAndCenter() : m_pt(m2::PointD()), m_featureId(0), m_rank(0) {}
+  FeatureWithRankAndCenter() : m_pt(m2::PointD()), m_id(0), m_rank(0) {}
 
-  FeatureWithRankAndCenter(m2::PointD pt, uint32_t featureId, uint8_t rank)
-    : m_pt(pt), m_featureId(featureId), m_rank(rank)
+  FeatureWithRankAndCenter(m2::PointD pt, uint32_t id, uint8_t rank)
```

Comment:
is it possible to pass m2::PointD by const ref?

Reply:
Surely.